### PR TITLE
[jdl] Export otherEntityField for every relationship (to side).

### PR DESCRIPTION
**Pull request by mshima**
Related to https://github.com/jhipster/generator-jhipster/pull/12551.

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.js [hunk 1]
```diff
@@ -132,20 +132,14 @@ function setRelationshipsToEntity(relatedRelationships, entityName) {
...
+        convertedRelationship.otherEntityField = lowerFirst(splitField.otherEntityField);
+        if (relationshipToConvert.type === ONE_TO_ONE || relationshipToConvert.type === MANY_TO_MANY) {
             convertedRelationship.ownerSide = false;
         } else if (relationshipToConvert.type === ONE_TO_MANY) {
             relationshipToConvert.injectedFieldInTo = relationshipToConvert.injectedFieldInTo || lowerFirst(relationshipToConvert.from);
-            convertedRelationship.relationshipName = camelCase(splitField.relationshipName || relationshipToConvert.from);
-            convertedRelationship.otherEntityField = lowerFirst(splitField.otherEntityField);
         } else if (relationshipToConvert.type === MANY_TO_ONE && relationshipToConvert.injectedFieldInTo) {
-            convertedRelationship.relationshipName = camelCase(splitField.relationshipName);
             convertedRelationship.relationshipType = 'one-to-many';
```
Comment (DanielFran) [inline jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.js:142]: Use the constant?

**Review by MathieuAA — APPROVED** [review]

File: jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.js (hunk 1 above)
Comment (mshima) [reply jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.js:142]:
```suggestion
            convertedRelationship.relationshipType = ONE_TO_MANY;
```

File: jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.js [hunk 2]
```diff
@@ -132,20 +132,14 @@ function setRelationshipsToEntity(relatedRelationships, entityName) {
...
-            convertedRelationship.relationshipName = camelCase(splitField.relationshipName || relationshipToConvert.from);
-            convertedRelationship.otherEntityField = lowerFirst(splitField.otherEntityField);
         } else if (relationshipToConvert.type === MANY_TO_ONE && relationshipToConvert.injectedFieldInTo) {
-            convertedRelationship.relationshipName = camelCase(splitField.relationshipName);
-            convertedRelationship.relationshipType = 'one-to-many';
-        } else if (relationshipToConvert.type === MANY_TO_MANY) {
-            convertedRelationship.relationshipName = camelCase(splitField.relationshipName);
-            convertedRelationship.otherEntityField = lowerFirst(splitField.otherEntityField);
-            convertedRelationship.ownerSide = false;
+            convertedRelationship.relationshipType = ONE_TO_MANY;
```
Comment (mshima) [inline jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.js:142]:
```suggestion
            convertedRelationship.relationshipType = 'one-to-many';
```

File: test/jdl/converters/JDLToJSON/jdl-with-applications-to-json-converter.spec.js [hunk 3]
```diff
@@ -1397,6 +1401,7 @@ describe('JDLWithApplicationsToJSONConverter', () => {
                                 databaseType: SQL,
                                 creationTimestamp: Date.now(),
                             });
+                            console.log(returned.get('toto'));
```
Comment (mshima) [inline test/jdl/converters/JDLToJSON/jdl-with-applications-to-json-converter.spec.js:1404]:
```suggestion
```

File: jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.js (hunk 1 above)
Comment (mshima) [reply jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.js:142]: `ONE_TO_MANY` is `OneToMany` instead of `one-to-many`

Comment (DanielFran) [reply jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.js:142]: This is strange!?! why to use different values for relationshipToConvert.type and convertedRelationship.relationshipType...

Comment (mshima) [reply jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.js:142]: relationshipToConvert.type is for jdl `relationship ManyToOne`, relationshipType is json `many-to-one`.

**Pull request merged**